tests/unit/testplan/testing/multitest/driver/myapp/test_app.py

```diff
@@ -326,3 +325,24 @@ def run_app(cwd, runpath):
     with app:
         app.proc.wait()
     return app
+
+
+def test_app_stop_after_shutdown(runpath):
+    """Test manually stopping an App after it has stopped."""
+    app = App(
+        name="App",
+        binary="echo",
+        args=["hello"],
+        shell=True,
+        runpath=runpath,
+    )
+    app.start()
+    app.wait(app.STATUS.STARTED)
+    assert app.status.tag == app.status.STARTED
+
+    app.stop()
```

Comment:
When the stop is called for the first time, self.proc is not None. And when the stop is called for the 2nd time, stop would see STOPPED status and return immediately. Thus I think the code change in App driver is not really exercised here? Is this testcase duplicate with the one for driver?